examples/hello/bpWriter/helloBPWriter_nompi.cpp

```diff
@@ -35,15 +35,17 @@ int main(int argc, char *argv[])
         adios2::Variable<float> bpFloats = bpIO.DefineVariable<float>(
             "bpFloats", {}, {}, {Nx}, adios2::ConstantDims);
 
+        std::string filename = "myVector_cpp.bp";
         /** Engine derived class, spawned to start IO operations */
         adios2::Engine bpWriter =
-            bpIO.Open("myVector_cpp.bp", adios2::Mode::Write);
+            bpIO.Open(filename, adios2::Mode::Write);
 
         /** Write variable for buffering */
         bpWriter.Put<float>(bpFloats, myFloats.data());
 
         /** Create bp file, engine becomes unreachable after this*/
         bpWriter.Close();
+        std::cout << "Wrote file " << filename << " to disk. It can now be read by running ./bin/hello_bpReader.\n";
     }
     catch (std::invalid_argument &e)
```

Comment:
Add an extra `catch(std::ios_base::failure &e) {  ..// you extra error message }`

Reply:
@williamfgc : The file already has a `catch(std::iOS_base::failure &e)`, so I tried to see if the error message was intelligible by removing read permissions from the file. But the error message was very intelligible:

```
./bin/hello_bpWriter
IO System base failure exception, STOPPING PROGRAM from rank 0
ERROR: couldn't open file myVector_cpp.bp, check permissions or path existence, in call to POSIX open
: unspecified iostream_category error
```

Presumably I could make this a bit better by going into `bpIO.Open` and doing a second check for permissions instead of the more broad message given here. Would you like that?

Reply:
@NAThompson you want to minimize requests to the file system in large scale code as it might compromise scalability. This is where adios is different from serial libraries. Users can live with this message, it's a 2 step check.